lib/adapters/xhr.js

```diff
@@ -28,6 +29,9 @@ module.exports = function xhrAdapter(config) {
     }
 
     var fullPath = buildFullPath(config.baseURL, config.url);
+    var parsed = url.parse(fullPath);
+    var protocol = parsed.protocol || 'http:';
```

Comment:
Considering the supported protocols are present in `lib/utils`, do you think this logic would be better isolated there?

```js
// Something similar to:
const getProtocol = ({ protocol }) => protocol || 'http:'
```

Reply:
Hello. Thanks for suggestion!